packages/flutter/lib/src/widgets/table.dart

```diff
@@ -417,6 +434,7 @@ class _TableElement extends RenderObjectElement {
       updateChildren(oldChildren, const <Widget>[], forgottenChildren: _forgottenChildren);
 
     _children = newChildren;
+    _doingMountOrUpdate = false;
```

Comment:
Why not set this flag at the end of the method, the `super.update(newWidget);` is not called here, Literally, the update has not been completed yet.